orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.groovy

```diff
@@ -135,30 +120,59 @@ class JedisExecutionRepository implements ExecutionRepository {
 
   @Override
   boolean isCanceled(String id) {
+    String key = fetchKey(id)
     withJedis { Jedis jedis ->
-      String key
-      if (jedis.exists("pipeline:$id")) {
-        key = "pipeline:$id"
-      } else if (jedis.exists("orchestration:$id")) {
-        key = "orchestration:$id"
-      } else {
-        throw new ExecutionNotFoundException("No execution found with id $id")
-      }
       Boolean.valueOf(jedis.hget(key, "canceled"))
     }
   }
 
   @Override
-  void updateStatus(String id, ExecutionStatus status) {
+  void pause(String id, String user) {
+    String key = fetchKey(id)
     withJedis { Jedis jedis ->
-      String key
-      if (jedis.exists("pipeline:$id")) {
-        key = "pipeline:$id"
-      } else if (jedis.exists("orchestration:$id")) {
-        key = "orchestration:$id"
-      } else {
-        throw new ExecutionNotFoundException("No execution found with id $id")
+      def currentStatus = ExecutionStatus.valueOf(jedis.hget(key, "status"))
+      if (currentStatus != ExecutionStatus.RUNNING) {
+        throw new IllegalStateException("Unable to pause pipeline that is not RUNNING (executionId: ${id}, currentStatus: ${currentStatus})")
       }
+
+      def pausedDetails = new Execution.PausedDetails(
+        pausedBy: user,
+        pauseTime: currentTimeMillis()
+      )
+
+      def data = [
+        paused: mapper.writeValueAsString(pausedDetails),
+        status: ExecutionStatus.PAUSED.toString()
+      ]
+      jedis.hmset(key, data)
+    }
+  }
+
+  @Override
+  void resume(String id, String user) {
+    String key = fetchKey(id)
+    withJedis { Jedis jedis ->
+      def currentStatus = ExecutionStatus.valueOf(jedis.hget(key, "status"))
+      if (currentStatus != ExecutionStatus.PAUSED) {
+        throw new IllegalStateException("Unable to resume pipeline that is not PAUSED (executionId: ${id}, currentStatus: ${currentStatus})")
+      }
+
+      def pausedDetails = mapper.readValue(jedis.hget(key, "paused"), Execution.PausedDetails)
```

Comment:
Should not be possible to be in PAUSED and not have `pausedDetails`. 

The `pause()` method on `JedisExecutionRepository` is the single source of setting the `PAUSED` status and does an atomic set of both `status` and `paused`.